lib/gui/app/modules/image-writer.js

```diff
@@ -126,16 +126,23 @@ exports.performWrite = (image, drives, onProgress) => {
       console.log(message)
     })
 
+    const flashResults = { errors: [] }
     ipc.server.on('fail', (error) => {
+      flashResults.errors.push(error)
       console.log('Fail:', error)
     })
 
-    const flashResults = {}
     ipc.server.on('done', (results) => {
       _.merge(flashResults, results)
     })
 
-    ipc.server.on('state', onProgress)
+    ipc.server.on('state', (progress) => {
+      flashResults.devices = {
+        succeeded: progress.succeeded,
+        failed: progress.failed
+      }
```

Comment:
Can we create this object once (and also init values to zero), and only update the values in this handler, to avoid thrashing the GC with all the created objects as well as potential ReferenceErrors (i.e. when accessing this before a state event happened)?